software/communication.py

```diff
@@ -0,0 +1,40 @@
+import serial
+import struct
+import serial.tools.list_ports
+
+
+
+class Communication:
+    def __init__(self):
+        self.ports=serial.tools.list_ports.comports()
+        for port in self.ports:
+            try:
+                self.testPort="/dev/"+port.name
+                self.testSerial=serial.Serial(self.testPort, baudrate=9600, timeout=2)
+                self.testSerial.write(struct.pack('<hhhHBH', 0,0,0,0,0,0xAAAA))
+                self.testRecvData=self.testSerial.read(8)
+                self.testData=struct.unpack('<hhhH', self.testRecvData)
+                self.testSerial.close()

```

Comment:
Why close the port? This point of code should only be reached by a valid serial connection, right?